File: Gevorgyan.TodoListApp.RestApi/ClientApp/src/app/core/models/TodoListViewModel.ts



// for AUTO-GENERATION!



/*
 * View model for entity "TodoList"
 */
export interface ITodoListViewModel {
    
    /*
     * Id
     */
    id: number;

    /*
     * Title
     */
    title: string;

    /*
     * Email of todo-list user
     */
    userEmail: string;
}
    
/*
 * View model for entity "TodoList"
 */
export class TodoListViewModel implements ITodoListViewModel {
    
    /*
     * Id
     */
    id: number;
    
    /*
     * Title
     */
    title: string;
    
    /*
     * Email of todo-list user
     */
    userEmail: string;
    
    constructor(data?: ITodoListViewModel) {
        
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            
            this.id = data["id"];
            this.title = data["title"];
            this.userEmail = data["userEmail"]; 
        }
    }

    static fromJS (data: any): TodoListViewModel {
        data = typeof data === 'object' ? data : {};
        let result = new TodoListViewModel;
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        
        data["id"] = this.id;
        data["title"] = this.title;
        data["userEmail"] = this.userEmail;
        
        return data; 
    }
}    

File: Gevorgyan.TodoListApp.Application/Queries/TodoLists/Management/Models/TodoListViewModel.cs
using AutoMapper;
using AutoMapper.Attributes;
using Gevorgyan.TodoListApp.Domain;

namespace Gevorgyan.TodoListApp.Application.Queries.TodoLists.Management.Models
{
    /// <summary>
    /// View model for entity "TodoList"
    /// </summary>
    [MapsFromTodoList]
    public class TodoListViewModel
    {
        /// <summary>
        /// Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Title 
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Email of todo-list user 
        /// </summary>
        public string UserEmail { get; set; }
    }

    /// <summary>
    /// Attribute for mapping from entity to model
    /// </summary>
    public class MapsFromTodoListAttribute : MapsFromAttribute
    {
        public MapsFromTodoListAttribute() : base(typeof(TodoList))
        {
        }

        public void ConfigureMapping(IMappingExpression<TodoList, TodoListViewModel> mappingExpression)
        {
            mappingExpression.ForMember(model => model.UserEmail, options => options.MapFrom(entity => entity.User.Email));
        }
    }
}
